src/Database/LSMTree/Internal.hs

```diff
@@ -530,13 +531,25 @@ toNormalLookupResult = \case
       Delete              -> Normal.NotFound
     Nothing -> Normal.NotFound
 
+{-# SPECIALISE updates :: V.Vector (SerialisedKey, Entry SerialisedValue SerialisedBlob) -> TableHandle IO h -> IO () #-}
 -- | See 'Database.LSMTree.Normal.updates'.
+--
+-- Does not enforce that mupsert and blobs should not occur in the same table.
 updates ::
-     forall m h k v blob.
-     V.Vector (k, Entry v blob)
+     m ~ IO -- TODO: replace by @io-classes@ constraints for IO simulation.
+  => V.Vector (SerialisedKey, Entry SerialisedValue SerialisedBlob)
   -> TableHandle m h
   -> m ()
-updates = error "updates: not yet implemented" -- TODO: implement
+updates es th = withOpenTable th $ \thEnv -> do
+    -- A placeholder implementation that is sufficient to pass the tests, but
+    -- keeps all entries in memory.
+    -- TODO: flush write buffer when full
+    -- TODO: merge runs when level becomes full
+    modifyMVar_ (tableWriteBuffer thEnv) $ \wb -> do
+      return $ foldl' (flip (uncurry (WB.addEntry resolve))) wb es
+  where
+    resolve =
+      maybe const unResolveMupsert (confResolveMupsert (tableConfig th))
```

Comment:
You can use https://github.com/IntersectMBO/lsm-tree/pull/251/files#diff-b75c6514218a4064abdf1339bbf6a6e163f0be909f3d5cb7602dbfe6312f6023R660-R661 here